engine/src/main/java/org/terasology/rendering/dag/tasks/SetCullFaceTask.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.tasks;
+
+import org.lwjgl.opengl.GL11;
+import org.terasology.rendering.dag.RenderPipelineTask;
+import org.terasology.rendering.dag.stateChanges.SetCullFace;
+
+/**
+ * TODO: Add javadocs
+ */
+public class SetCullFaceTask implements RenderPipelineTask {
+    private int modeValue;
+    private SetCullFace.Mode mode;
+
+    public SetCullFaceTask(SetCullFace.Mode mode) {
```

Comment:
As you switch back to OpenGL integer constants rather than enums, this whole method will be drastically simplified as you'll be able to rely on one of the three correct integers coming through. However, you will need some kind of integer to meaningful string translation for the toString() method.